pandas/core/arrays/string_arrow.py

```diff
@@ -663,7 +664,10 @@ def _convert_int_dtype(self, result):
         return result
 
     def _cmp_method(self, other, op):
-        result = super()._cmp_method(other, op)
+        try:
```

Comment:
I think after https://github.com/pandas-dev/pandas/pull/56003 this won't be needed

Reply:
why? Will this make comparing strings to any non strings less strict?

Reply:
This is still needed